forge/src/main/java/org/openjdk/skara/forge/bitbucket/BitbucketRepository.java

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.skara.forge.bitbucket;
+
+import java.net.URI;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import org.openjdk.skara.forge.Check;
+import org.openjdk.skara.forge.CommitComment;
+import org.openjdk.skara.forge.Forge;
+import org.openjdk.skara.forge.HostedBranch;
+import org.openjdk.skara.forge.HostedCommit;
+import org.openjdk.skara.forge.HostedRepository;
+import org.openjdk.skara.forge.PullRequest;
+import org.openjdk.skara.forge.WebHook;
+import org.openjdk.skara.forge.WorkflowStatus;
+import org.openjdk.skara.host.HostUser;
+import org.openjdk.skara.issuetracker.Label;
+import org.openjdk.skara.json.JSONValue;
+import org.openjdk.skara.network.URIBuilder;
+import org.openjdk.skara.vcs.Branch;
+import org.openjdk.skara.vcs.Hash;
+import org.openjdk.skara.vcs.ReadOnlyRepository;
+import org.openjdk.skara.vcs.Tag;
+import org.openjdk.skara.vcs.VCS;
+
+public class BitbucketRepository implements HostedRepository {
+    private final BitbucketHost bitbucketHost;
```

Comment:
I would have named the field just `host`.

Reply:
Right, again copy paste and just mindlessly modifying the field names. This I will adjust.